### PR TITLE
Convert communication q_due_date to user's TZ

### PR DESCRIPTION
**Pull request by OptimusRhine**
https://www.pivotaltracker.com/story/show/151540337

* modified `questionnaire_due_date` for LR-generated Communication emails, to convert from UTC to the user's TZ (as that may modify the due date by +/- 1), if user TZ is set
* added tests for `questionnaire_due_date` (without and with user TZ)
  * added new `setdate` param to `bless_with_basics()`, to make sure we can test a time where converting to another timezone would affect the day

### CODE REVIEW

File: portal/models/communication.py [hunk 1]
```diff
@@ -127,7 +128,13 @@ def _lookup_questionnaire_due_date():
         trigger_date = qb.trigger_date(user)
         due = (qb.calculated_overdue(trigger_date) or
                qb.calculated_expiry(trigger_date))
-        return due.strftime('%-d %b %Y') if due else ''
+        due_date = ''
+        if user.timezone and due:
+            local = utc.localize(due).astimezone(timezone(user.timezone))
```
Comment (pbugni) [inline portal/models/communication.py]: Can we make this a function on user or perhaps better in `date_tools`?  I would expect there are several places where we'll need to return the timezone adjusted date / time.

Comment (OptimusRhine) [reply portal/models/communication.py]: Sure thing

**Comment by OptimusRhine** [comment]
@pbugni Should the `intervention_strategies` fix be part of this PR? Or should we go ahead and get this change in, then add that separately?

**Comment by pbugni** [comment]
@OptimusRhine fair - let's file as a separate issue.  Merging.

**Pull request merged**